Source/WebKit/WebProcess/WebPage/dmabuf/AcceleratedSurfaceDMABuf.cpp

```diff
@@ -606,18 +632,26 @@ void AcceleratedSurfaceDMABuf::didRenderFrame(const std::optional<WebCore::Regio
     if (!m_target)
         return;
 
-    if (auto fence = WebCore::GLFence::create(WebCore::GLFence::ShouldFlush::No))
-        fence->wait(WebCore::GLFence::FlushCommands::Yes);
-    else
+    UnixFileDescriptor didRenderFrameFence;
+    if (m_useExplicitSync && m_target->supportsExplicitSync()) {
+        auto& display = WebCore::PlatformDisplay::sharedDisplayForCompositing();
+        EGLSyncKHR fence = eglCreateSyncKHR(display.eglDisplay(), EGL_SYNC_NATIVE_FENCE_ANDROID, nullptr);
+        glFlush();
+        if (fence != EGL_NO_SYNC_KHR) {
+            didRenderFrameFence = UnixFileDescriptor { eglDupNativeFenceFDANDROID(display.eglDisplay(), fence), UnixFileDescriptor::Adopt };
+            eglWaitSyncKHR(display.eglDisplay(), fence, 0);
+            eglDestroySyncKHR(display.eglDisplay(), fence);
+        }
+    } else
         glFlush();
```

Comment:
When explicit sync is not available or not supported by the target or we fail to create the fence we should use a normal fence and wait like we did before. Again, maybe it would be easier to handle the fence in the target adding `RenderTarget::createRenderingFence`